src/System.Private.ServiceModel/tools/test/SelfHostWcfService/TestResources/RequestLoggingServiceBehavior.cs

```diff
@@ -0,0 +1,138 @@
+        public RequestLoggingServiceBehavior() { }
+
+        public RequestLoggingServiceBehavior(LogRequest logger)
+        {
+            _logCallback = logger;
+        }
+
+        public void AddLogger(LogRequest logger)
+        {
+            _logCallback += logger;
+        }
+
+        public void Validate(ServiceDescription serviceDescription, ServiceHostBase serviceHostBase) { }
+
+        public void AddBindingParameters(ServiceDescription serviceDescription, ServiceHostBase serviceHostBase, Collection<ServiceEndpoint> endpoints,
+            BindingParameterCollection bindingParameters)
+        { }
+
+        public void ApplyDispatchBehavior(ServiceDescription serviceDescription, ServiceHostBase serviceHostBase)
+        {
+            serviceHostBase.Authorization.ServiceAuthorizationManager =
+                new RequestLoggingAuthenticationWrapper(this, serviceHostBase.Authorization.ServiceAuthorizationManager);
+            foreach (ChannelDispatcherBase t in serviceHostBase.ChannelDispatchers)
+            {
+                ChannelDispatcher channelDispatcher = t as ChannelDispatcher;
+                foreach (EndpointDispatcher endpointDispatcher in channelDispatcher.Endpoints)
+                {
+                    endpointDispatcher.DispatchRuntime.ServiceAuthorizationManager =
+                        new RequestLoggingAuthenticationWrapper(this,
+                            endpointDispatcher.DispatchRuntime.ServiceAuthorizationManager);
+                }
+            }
+        }
+
+        private void LogIncomingRequest(OperationContext operationContext, ref Message message, bool authenticated)
+        {
+            try
+            {
+                if (_logCallback != null)
+                {
+                    _logCallback(operationContext, ref message, authenticated);
+                }
+            }
+            catch{ /* Swallow any exceptions */ }
+        }
+
+        public static void SimpleConsoleLogger(OperationContext operationContext, ref Message message, bool authenticated)
+        {
+            var log = new StringBuilder();
+            log.Append("Via:");
+            object requestMessagePropertyObject;
+            if (operationContext.IncomingMessageProperties.TryGetValue(HttpRequestMessageProperty.Name,
+                out requestMessagePropertyObject))
+            {
+                var requestMessageProperty = requestMessagePropertyObject as HttpRequestMessageProperty;
+                if (requestMessageProperty != null)
+                {
+                    log.Append(requestMessageProperty.Method).Append(' ');
+                }
+            }
+
+            log.AppendLine(message.Properties.Via.ToString());
+            log.Append("To Header:").AppendLine(message.Headers.Action);
+            log.Append("Authentication ").AppendLine(authenticated ? "succeeded" : "failed");
+            Console.WriteLine(log.ToString());
```

Comment:
Not for this PR, but at some point we need have a central mechanism for logging or tracing from the Bridge.  As written, we will see this spew but not know where it is coming from or why.